src/mutable_data.rs

```diff
@@ -133,3 +681,103 @@ impl MutableDataRef {
         self.tag
     }
 }
+
+#[derive(Debug, Clone)]
+pub enum SeqEntryAction {
+    /// Inserts a new Sequenced entry
+    Ins(Value),
+    /// Updates an entry with a new value and version
+    Update(Value),
+    /// Deletes an entry
+    Del(u64),
+}
+
+#[derive(Debug, Clone)]
+pub enum UnseqEntryAction {
+    /// Inserts a new Unsequenced entry
+    Ins(Vec<u8>),
+    /// Updates an entry with a new value
+    Update(Vec<u8>),
+    /// Deletes an entry
+    Del,
+}
+
+/// Helper struct to build entry actions on `MutableData`
+#[derive(Debug, Default, Clone)]
+pub struct SeqEntryActions {
+    actions: BTreeMap<Vec<u8>, SeqEntryAction>,
+}
+
+impl SeqEntryActions {
+    /// Create a helper to simplify construction of `MutableData` actions
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Insert a new key-value pair
+    pub fn ins(mut self, key: Vec<u8>, content: Vec<u8>, version: u64) -> Self {
+        let _ = self.actions.insert(
+            key,
+            SeqEntryAction::Ins(Value {
+                data: content,
+                version,
+            }),
+        );
+        self
+    }
+
+    /// Update an existing key-value pair
+    pub fn update(mut self, key: Vec<u8>, content: Vec<u8>, version: u64) -> Self {
+        let _ = self.actions.insert(
+            key,
+            SeqEntryAction::Update(Value {
+                data: content,
+                version,
+            }),
+        );
+        self
+    }
+
+    /// Delete an entry
+    pub fn del(mut self, key: Vec<u8>, version: u64) -> Self {
+        let _ = self.actions.insert(key, SeqEntryAction::Del(version));
+        self
+    }
+}
+
+impl Into<BTreeMap<Vec<u8>, SeqEntryAction>> for SeqEntryActions {
+    fn into(self) -> BTreeMap<Vec<u8>, SeqEntryAction> {
+        self.actions
+    }
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct UnseqEntryActions {
+    actions: BTreeMap<Vec<u8>, UnseqEntryAction>,
+}
+
+impl UnseqEntryActions {
+    /// Insert a new key-value pair
+    pub fn ins(mut self, key: Vec<u8>, content: Vec<u8>) -> Self {
+        let _ = self.actions.insert(key, UnseqEntryAction::Ins(content));
+        self
+    }
+
+    /// Update existing key-value pair
+    pub fn update_unseq(mut self, key: Vec<u8>, content: Vec<u8>) -> Self {
```

Comment:
Probably it'd be better to have a uniform API here?
I'd suggest using just `update` here, but ultimately I'd leave this to your judgement.